compiler/src/diagnostics/output.re

```diff
@@ -43,9 +55,36 @@ let exn_to_lsp_error = (exn: exn): option(lsp_error) => {
 // kept lenses as an empty array so older LSPs don't crash
 
 let result_to_json =
-    (~errors: list(lsp_error), ~values: list(Lenses.lens_t)) => {
-  let result: lsp_result = {errors, values, lenses: []};
+    (
+      ~errors: list(lsp_error),
+      ~warnings: list(lsp_warning),
+      ~values: list(Lenses.lens_t),
+    ) => {
+  let result: lsp_result = {errors, warnings, values, lenses: []};
   Yojson.Basic.pretty_to_string(
     Yojson.Safe.to_basic(lsp_result_to_yojson(result)),
   );
 };
+
+let convert_warnings = (warnings_this_run, topLevelFileName) => {
+  List.map(
```

Comment:
@ospencer Do you think that warnings emitted at the dummy location should be included here?

Reply:
I just merged main on this PR, so this refers to a change that went through #659